### PR TITLE
[Feature](insert-overwrite) Support create partition for auto partition table when insert overwrite

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

introduced session variable `enable_auto_create_when_overwrite`

when it's true:
1. `insert overwrite table auto_partition_table [values xxx| select xxx]` support overwrite old datas and create partition(s) for new datas if need.
2. `insert overwrite table auto_partition_table partition(*) [values xxx| select xxx]` support overwrite old datas for values-relative partitions(as it was before) and create partition(s) for new datas if need.

doc pr: https://github.com/apache/doris-website/pull/936

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run compile

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/sink/vrow_distribution.cpp [hunk 1]
```diff
@@ -456,15 +507,19 @@ Status VRowDistribution::generate_rows_distribution(
...
+        st = _generate_rows_distribution_for_auto_overwrite(block.get(), partition_cols_idx,
+                                                            has_filtered_rows, row_part_tablet_ids,
+                                                            rows_stat_val);
     } else if (_vpartition->is_auto_partition() && !_deal_batched) {
+        VLOG_TRACE << "Debugging " << this << '\n' << block->dump_data(0, 1) << "auto partition";
         st = _generate_rows_distribution_for_auto_partition(block.get(), partition_cols_idx,
                                                             has_filtered_rows, row_part_tablet_ids,
                                                             rows_stat_val);
     } else { // not auto partition
+        VLOG_TRACE << "Debugging " << this << '\n' << block->dump_data(0, 1) << "normal batch";
```
Comment (HappenLee) [inline be/src/vec/sink/vrow_distribution.cpp:522]: Remove the Debuggging log

File: be/src/vec/sink/vrow_distribution.cpp [hunk 2]
```diff
@@ -456,15 +507,19 @@ Status VRowDistribution::generate_rows_distribution(
     }
 
     Status st = Status::OK();
-    if (_vpartition->is_auto_detect_overwrite()) {
+    if (_vpartition->is_auto_detect_overwrite() && !_deal_batched) {
```
Comment (HappenLee) [inline be/src/vec/sink/vrow_distribution.cpp:510]: the code wired. seems the `deal_batched` should always be true in call func `_generate_rows_distribution_for_auto_overwrite`

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM BE code

**Review by zddr — APPROVED** [review]
LGTM

**Pull request merged**